### PR TITLE
Fix allocations with matrix-vector products

### PR DESCRIPTION
**Pull request by geoffroyleconte**

### CODE REVIEW

**Review by dpo — APPROVED** [review]

File: src/coo_linalg.jl [hunk 1]
```diff
@@ -11,13 +25,24 @@ function LinearAlgebra.mul!(
...
 end
 
+function coo_adjtrans_mul!(C::AbstractVector, Arows, Acols, Avals, B::AbstractVector, α, Annz, t)
+  @inbounds for k = 1:Annz
+    i, j = Acols[k], Arows[k] 
+    C[i] += α * t(Avals[k]) * B[j]
+  end
+end
+
+function coo_adjtrans_mul!(C::AbstractMatrix, Arows, Acols, Avals, B::AbstractMatrix, α, Annz, t)
```
Comment (tmigot) [inline src/coo_linalg.jl:39]: Can you add a unit test for this one to increase the coverage?

**Review by tmigot — APPROVED** [review]
Apart for a small comment, it looks great for me. Thanks @geoffroyleconte !

File: src/coo_linalg.jl [hunk 2]
```diff
@@ -11,13 +25,24 @@ function LinearAlgebra.mul!(
   if β != 1
     β != 0 ? rmul!(C, β) : fill!(C, zero(eltype(C)))
   end
-  @inbounds for k = 1:nnz(A)
-    i, j = A.rows[k], A.cols[k]
-    @views C[i, :] += α * A.vals[k] * B[j, :]
```
[the remark this answers is not in the document]
Comment (geoffroyleconte) [reply src/coo_linalg.jl:16]: Yes, you need to use the broadcast

[the remark this answers is not in the document]
Comment (dpo) [reply src/coo_linalg.jl:16]: Great. Thanks for catching that!

**Pull request merged**